### ISSUE TITLE
Inaccurate error message for return type of async method

### ISSUE THREAD

**Issue by svick** [dotnet/roslyn#17417]
**Version Used**: VS2017RC (VisualStudio/15.0.0-RC.4+26206.0)

**Steps to Reproduce**:

Create a new .Net Framework Class Library with the following code:

```c#
class C
{
    async NotTask Func() => 0;
}

class NotTask { }
```

**Actual Behavior**: An error message that says:

> CS1983	The return type of an async method must be void, Task or Task\<T>

This is inaccurate, because task-like types (like `ValueTask<T>`) can be returned from `async` methods in C# 7.0.

**Expected Behavior**: An error message that includes task-like types, possibly:

> The return type of an async method must be void, Task, Task\<T> or task-like

**Comment by AdamSpeight2008** [comment]
I remember this being brought up before, even being implemented.

**Comment by jaredpar** [comment]
Thanks for reporting.  Looks like a dupe of #12621
